src/toxcore/dht_new/packet.rs

```diff
@@ -0,0 +1,741 @@
+            //DhtPacket::CookieResp(ref p) => p.to_bytes(buf),
+            //DhtPacket::CryptoHs(ref p) => p.to_bytes(buf),
+            //DhtPacket::CryptoData(ref p) => p.to_bytes(buf),
+            DhtPacket::DhtRequest(ref p) => p.to_bytes(buf),
+            //DhtPacket::LanDisc(ref p) => p.to_bytes(buf),
+            // DhtPacket::OnionReq0(ref p) => p.to_bytes(buf),
+            // DhtPacket::OnionReq1(ref p) => p.to_bytes(buf),
+            // DhtPacket::OnionReq2(ref p) => p.to_bytes(buf),
+            //DhtPacket::AnnReq(ref p) => p.to_bytes(buf),
+            //DhtPacket::AnnResp(ref p) => p.to_bytes(buf),
+            // DhtPacket::OnionDataReq(ref p) => p.to_bytes(buf),
+            // DhtPacket::OnionDataResp(ref p) => p.to_bytes(buf),
+            // DhtPacket::OnionResp3(ref p) => p.to_bytes(buf),
+            // DhtPacket::OnionResp2(ref p) => p.to_bytes(buf),
+            // DhtPacket::OnionResp1(ref p) => p.to_bytes(buf),
+        }
+    }
+}
+/// De-serialize bytes into `DhtPacket`.
+impl FromBytes for DhtPacket {
+    named!(from_bytes<DhtPacket>, alt!(
+        map!(PingReq::from_bytes, DhtPacket::PingReq) |
+        map!(PingResp::from_bytes, DhtPacket::PingResp) |
+        map!(GetNodes::from_bytes, DhtPacket::GetNodes) |
+        map!(SendNodes::from_bytes, DhtPacket::SendNodes) |
+        //map!(CookieReq::from_bytes, DhtPacket::CookieReq) |
+        //map!(CookieResp::from_bytes, DhtPacket::CookieResp) |
+        //map!(CryptoHs::from_bytes, DhtPacket::CryptoHs) |
+        //map!(CryptoData::from_bytes, DhtPacket::CryptoData) |
+        map!(DhtRequest::from_bytes, DhtPacket::DhtRequest)
+        //map!(LanDisc::from_bytes, DhtPacket::LanDisc) |
+        // map!(OnionReq0::from_bytes, DhtPacket::OnionReq0) |
+        // map!(OnionReq1::from_bytes, DhtPacket::OnionReq1) |
+        // map!(OnionReq2::from_bytes, DhtPacket::OnionReq2) |
+        //map!(AnnReq::from_bytes, DhtPacket::AnnReq) |
+        //map!(AnnResp::from_bytes, DhtPacket::AnnResp)
+        // map!(OnionDataReq::from_bytes, DhtPacket::OnionDataReq) |
+        // map!(OnionDataResp::from_bytes, DhtPacket::OnionDataResp) |
+        // map!(OnionResp3::from_bytes, DhtPacket::OnionResp3) |
+        // map!(OnionResp2::from_bytes, DhtPacket::OnionResp2) |
+        // map!(OnionResp1::from_bytes, DhtPacket::OnionResp1)
+    ));
+}
+
+/// Length in bytes of [`PingReq`](./struct.PingReq.html) and
+/// [`PingResp`](./struct.PingResp.html) when serialized into bytes.
+pub const PING_SIZE: usize = 9;
+
+
+/**
+Used to request/respond to ping. Use in an encrypted form.
+
+Used in:
+
+- [`DhtPacket`](./struct.DhtPacket.html)
+- [`DhtRequest`](./struct.DhtRequest.html)
+
+Serialized form:
+
+Ping Packet (request and response)
+
+Packet type `0x00` for request and `0x01` for response.
+
+Response ID must match ID of the request, otherwise ping is invalid.
+
+Length      | Contents
+----------- | --------
+`1`         | `u8` packet type
```

Comment:
instead of   `u8` packet type put down this packet type. So the line will look like:
```
`1`         | 0x00
```

And do it for all other packet types